cmd/gardenlet/app/app.go

```diff
@@ -131,11 +133,7 @@ func NewCommand() *cobra.Command {
 }
 
 func run(ctx context.Context, cancel context.CancelFunc, log logr.Logger, cfg *config.GardenletConfiguration) error {
-	// Add feature flags
-	if err := gardenletfeatures.FeatureGate.SetFromMap(cfg.FeatureGates); err != nil {
-		return err
-	}
```

Comment:
Nit: Doesn't this `SetFromMap` invocation still better belong here than in `validate`?

Reply:
Calling `SetFromMap` in `run` is too late. The config file validation will complain about the IPv6 `seedConfig` if the `IPv6SingleStack` feature gate is not enabled (via `SetFromMap`) beforehand.
I improved the comment to make this clearer and performed the change consistently across all component entry points.